### DISCUSSION TITLE
Please.Build LSP

### DISCUSSION THREAD

**Discussion by Dmitry-N-Medvedev** [LazyVim/LazyVim#4973] [Q&A] [answered]
good morning, nice Lazy community :)

**the context**: I would like to install Please.Build LSP server to LazyVim.
here is the snippet: https://github.com/neovim/nvim-lspconfig/blob/master/doc/configs.md#please

```require'lspconfig'.please.setup{}```

**the problem**: I have no idea which file I should add this snippet to.

any help is greatly appreciated :)

best regards,
Dmitry

**Comment by dpetka2001** [comment, marked as answer]
Just create a file in `~/.config/nvim/lua/plugins/` directory with contents
```lua
return {
  {
    "neovim/nvim-lspconfig",
    opts = {
      servers = {
        please = {},
      
    },
  },
}
```
You will also have to have a `.plzconfig` file at the root of your project according to the [docs](https://github.com/neovim/nvim-lspconfig/blob/b8b725659fa60d2c5f5bd7459bcfee1d8b34acd5/lua/lspconfig/configs/please.lua#L7) for the server to attach to your files.

**Reply by Dmitry-N-Medvedev** [reply]
cool! this worked with a caveat:

I had to do `:set filetype=bzl` while having a BUILD.plz file opened to confirm the solution works.

not sure if this is the correct place to ask another question, please let me know if it is not the case.

the additional question: into which configuration files should I place the `:set filetype=bzl` instruction, so that the *.plz files are handled by the Please.Build LSP?

**Reply by dpetka2001** [reply]
Just add a file `~/.config/nvim/filetype.lua` with contents
```lua
vim.filetype.add({
  pattern = {
    [".*%.plz"] = "bzl",
  }
})
```
I believe that should do it. Never used this server myself and not familiar with it, so please test it out.

**Reply by Dmitry-N-Medvedev** [reply]
this has worked out perfectly fine :) thank you :)

a tiny note: I have put your code to ~/.config/nvim/lua/config/autocmds.lua file
